### ISSUE TITLE
requirement.txt导入问题

### ISSUE THREAD

**Issue by BBKing926** [greekinRoma/SMPISD-MTPNet#2]
按照提示进行创建环境导入发生错误，自己pip 安装也不对，求指教应该如何进行正确安装

**Comment by greekinRoma** [comment]
同学，requirement.txt的代码已经更新，如果还是有问题，请再联系我

**Comment by greekinRoma** [comment]
import numpy as np
import cv2
import matplotlib.pyplot as plt
from sklearn.metrics import auc


class ROCMetric(object):
    def __init__(self, bins=100):
        self.bins = bins
        self.reset()

    def update(self, pred, label):
        pred = pred / np.max(pred) # normalize output to 0-1
        label = label.astype(np.uint8)

        # analysis target number
        num_labels, labels, _, centroids = cv2.connectedComponentsWithStats(label)
        assert num_labels > 1

        # get masks and update background area and targets number
        back_mask = labels == 0
        tmp_back_area = np.sum(back_mask)
        self.background_area += tmp_back_area
        self.target_nums += (num_labels - 1)

        for ibin in range(self.bins + 1):
            thre = ibin / self.bins
            pred_binary = pred >= thre

            # update false detection
            tmp_false_detect = np.sum(np.logical_and(back_mask, pred_binary))
            assert tmp_false_detect <= tmp_back_area
            self.false_detect[ibin] += tmp_false_detect

            # update true detection, there maybe multiple targets
            for t in range(1, num_labels):
                target_mask = labels == t
                self.true_detect[ibin] += np.sum(np.logical_and(target_mask, pred_binary)) > 0

    def get(self):
        fpr = self.false_detect / self.background_area  # X axis
        tpr = self.true_detect / self.target_nums       # Y axis
        return fpr, tpr, auc(fpr, tpr)

    def get_all(self):
        return self.false_detect, self.background_area, self.true_detect, self.target_nums

    def reset(self):
        self.false_detect = np.zeros(self.bins+1)
        self.true_detect = np.zeros(self.bins+1)
        self.background_area = 0
        self.target_nums = 0

**Comment by greekinRoma** [comment]
我公开的代码里面有的哈，在evaluation文件夹中

**Comment by BBKing926** [comment]
> 同学，requirement.txt的代码已经更新，如果还是有问题，请再联系我

你好，作者，我已经成功安装requirement文件，谢谢你，但我现在遇到的问题是，下载好的数据集是应该使用COCO中的SDSS吗，然后按照6：1：3划分，然后应该将数据集放在哪个目录下，我没有找到相关的路径

**Comment by BBKing926** [comment]
你好作者，麻烦问一下下载好的数据集是应该使用COCO中的SDSS吗，然后按照6：1：3划分，然后应该将数据集放在哪个目录下，图像，掩码还有标注怎么放呢，我没有找到相关的路径




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "greekinRoma/SMPISD-MTPNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年6月4日(星期三) 下午4:10
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [greekinRoma/SMPISD-MTPNet] requirement.txt导入问题 (Issue #2)



greekinRoma left a comment (greekinRoma/SMPISD-MTPNet#2)
 
我公开的代码里面有的哈，在evaluation文件夹中

**Comment by greekinRoma** [comment]
> 你好作者，麻烦问一下下载好的数据集是应该使用COCO中的SDSS吗，然后按照6：1：3划分，然后应该将数据集放在哪个目录下，图像，掩码还有标注怎么放呢，我没有找到相关的路径
> […](#)

抱歉这么久才回复。这个代码是我一年半之前写的，我也记不太清楚了。但是你可以follow mask source.py这个文件应该可以找到对应的位置。
